src/main/scala/microsites/MicrositeKeys.scala

```diff
@@ -167,14 +201,64 @@ trait MicrositeAutoImportSettings extends MicrositeKeys {
 
   lazy val fr = new FileReader
 
+  def createVersionsJson(targetDir: String, content: List[Version]): File = {
+    val jekyllDir  = "jekyll"
+    val targetPath = s"$targetDir$jekyllDir/_data/versions.json"
+    createFile(targetPath)
+    writeContentToFile(content.asJson.toString, targetPath)
+    targetPath.toFile
+  }
+
+  def generateVersionList(versionStringList: List[String], ownVersion: String) = {
+    versionStringList
+      .map(version => {
+        Version(version, own = (ownVersion == version))
+      })
+      .toList
+  }
+
+  def pluginProjects(pluginName: String): Option[Array[String]] = {
+    val sbtPluginsOutput = "sbt --error plugins".lineStream
+    val pluginLine =
+      sbtPluginsOutput.find(line => line.trim.startsWith(s"$pluginName: enabled in "))
+
+    pluginLine match {
+      case Some(line) => Some(line.trim.stripPrefix(s"$pluginName: enabled in ").split(", "))
+      case None       => None
+    }
+  }
+
+  // Generate a microsite externally through sbt and sbt-microsites tasks
+  def createMicrositeVersion(
+      sourceDir: String,
+      targetDir: String,
+      baseUrl: String,
+      version: String): Unit = {
+    val newBaseUrl =
+      if (version != "") s"${baseUrl}/$version" else s"${baseUrl}"
+    val pluginName            = "microsites.MicrositesPlugin"
+    val sbtMicrositesProjects = pluginProjects(pluginName)
+    sbtMicrositesProjects match {
+      case Some(projects) => {
+        List(
+          "sbt",
+          s"""clean; set ${projects(0)}/micrositeBaseUrl := "$newBaseUrl"; ${projects(0)}/makeMicrosite""").!
+        Files.move(
+          FileSystems.getDefault().getPath(sourceDir),
+          FileSystems.getDefault().getPath(s"$targetDir/$version"),
```

Comment:
I believe you can use:

```scala
Paths.get(pathString)
```

in both cases above.

Reply:
Addressed in https://github.com/47deg/sbt-microsites/pull/379/commits/41e996fd7cf2934d9ab7d07d58d722506ab70b9b